tools/dev-doctor/rootdircheck.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+)
+
+// A rootDirCheck checks that Cilium is checked out in
+// ${GOPATH}/src/github.com/cilium/cilium.
+type rootDirCheck struct{}
+
+func (rootDirCheck) Name() string {
+	return "root-dir"
+}
+
+func (rootDirCheck) Run() (checkResult, string) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return checkError, fmt.Sprintf("cannot get working directory: %s", err)
+	}
+
+	// Search upward through through parent directories to find the .git directory.
+	for {
+		info, err := os.Stat(path.Join(dir, ".git"))
+		switch {
+		case err == nil && info.Mode().IsDir():
+			if dir != path.Join(os.Getenv("GOPATH"), "src", "github.com", "cilium", "cilium") {
```

Comment:
if `GOPATH==""` this could succeed even though in reality wouldn't work, right? Perhaps we could check for  `!!GOPATH` first

Reply:
Thanks, done.